tests/flow/test_config.py

```diff
@@ -168,5 +176,41 @@ def test08_config_restore_timeout_to_default(self):
         self.env.assertEqual(response, expected_response)
 
         # Issue long-running query to validate the reconfiguration
-        result = redis_graph.query("UNWIND range(1,1000000) AS v RETURN COUNT(v)")
+        result = redis_graph.query(query)
         self.env.assertEqual(result.result_set[0][0], 1000000)
+
+    def test09_set_invalid_values(self):
+        # The run-time configurations supported by RedisGraph are:
+        # MAX_QUEUED_QUERIES
+        # TIMEOUT
+        # QUERY_MEM_CAPACITY
+        # DELTA_MAX_PENDING_CHANGES
+        # RESULTSET_SIZE
+
+        # Validate that attempting to set these configurations to
+        # invalid values fails
+        try:
+            # MAX_QUEUED_QUERIES must be a positive value
+            redis_con.execute_command("GRAPH.CONFIG SET MAX_QUEUED_QUERIES 0")
+            assert(False)
+        except redis.exceptions.ResponseError as e:
+            assert("Failed to set config value MAX_QUEUED_QUERIES to 0" in str(e))
+            pass
+
+        # TIMEOUT, QUERY_MEM_CAPACITY, and DELTA_MAX_PENDING_CHANGES must be
+        # non-negative values, 0 resets to default
+        for config in ["TIMEOUT", "QUERY_MEM_CAPACITY", "DELTA_MAX_PENDING_CHANGES"]:
+            try:
+                redis_con.execute_command("GRAPH.CONFIG SET %s -1" % config)
+                assert(False)
+            except redis.exceptions.ResponseError as e:
+                assert("Failed to set config value %s to -1" % config in str(e))
+                pass
+
+        # RESULTSET_SIZE can be any integer, negative values reset to default
```

Comment:
Can you please point me to where do we treat negative values as reverting to default in this case ?

Reply:
https://github.com/RedisGraph/RedisGraph/blob/master/src/configuration/config.c#L216

Reply:
But resultset_size is defined as `uint64_t resultset_size` seems like it would never be negative

Reply:
The config, like all others, are read from Redis as a `long long`. It's then cast to an `int64_t` when calling `Config_resultset_max_size_set`, then set as a `uint64_t` when saved in the struct - `UINT64_MAX` if negative.

Do you want me to make 0 reset to the default to make this more consistent with the others?

Reply:
No that's fine, 0 is a valid result-set size, although I don't think anyone will use it, also as you've said this diverts from most other configurations.